src/ResourceManager/Network/Commands.Network/ExpressRouteCircuit/ExpressRouteCircuitBaseCmdlet.cs

```diff
@@ -72,5 +75,30 @@ public PSExpressRouteCircuit ToPsExpressRouteCircuit(Management.Network.Models.E
 
             return psCircuit;
         }
+
+        public List<ExpressRouteCircuit> GetAllResourcesByPollingNextLink(IPage<ExpressRouteCircuit> resourcePage)
+        {
+            var resourceList = new List<ExpressRouteCircuit>();
+
+            var nextPageLink = this.AddResourceToListAndReturnNextPageLink(resourcePage, resourceList);
+
+            while (!string.IsNullOrEmpty(nextPageLink))
```

Comment:
Wouldn't it be simpler to create a generic enumerator over IPage<t> that returned an IEnumerable<T>?  You can find a sample of this code here: 9just needs to be made generic):

https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Generated/Models/PageEnumerator.cs 

https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Generated/Models/PageEnumerablecs